### PR TITLE
Added validation

### LINKED ISSUE #5: Thumbnail in category
What this field means?
I can write any string in admin on category edit page.

### LINKED ISSUE #6: Generating slug by title
Add ability to generate slug based on title after create new category/post if field is empty.

### LINKED ISSUE #8: Exception when I'm trying create Post without thumbnail
```
Catchable Fatal Error: Argument 1 passed to Application\BlogBundle\Service\Thumbnail::save() must be an instance of Symfony\Component\HttpFoundation\File\UploadedFile, null given, called in /Users/aivus/projects/alexeySymfonyApp/src/Application/BlogBundle/Form/DataTransformer/UploadedFileToThumbnail.php on line 31 and defined
```

### LINKED ISSUE #10: Field "Content" has asteriks, but not required
Got exception
```
Failed to create object: Application\BlogBundle\Entity\Post
```

### LINKED ISSUE #12: Add validation
Add validation for fields.
Example: length validation.

At least, see on a entity configuration and add validation for correct handling

### LINKED ISSUE #25: Fatal error when slug already used
```
[3/4] UniqueConstraintViolationException: An exception occurred while executing 'INSERT INTO post (title, slug, published, thumbnail, description, keywords, body, published_at, created_at, updated_at, category_id) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' with params ["sadasdsa", "sadasd", 0, false, null, null, "<p>sadasdasdasd<\/p>", null, "2016-03-11 17:27:15", "2016-03-11 17:27:15", 1]:

SQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: post.slug   +
```

### PR DESCRIPTION
**Pull request by Madlexx**
fixed #12 
fixed #5 
fixed #8 
fixed #6 
fixed #25

### CODE REVIEW

**Comment by aivus** [comment]
@Madlexx, please don't merge "all in one".
It's very difficult to understand.

**Comment by aivus** [comment]
I removed commit for entity fix from this branch

File: src/Application/BlogBundle/Admin/CategoryAdmin.php [hunk 1]
```diff
@@ -14,11 +15,25 @@ class CategoryAdmin extends Admin
...
     {
-        $formMapper->add('title', null, ['label' => 'blog_bundle.property.title'])
-            ->add('slug', null, ['label' => 'blog_bundle.property.slug'])
-            ->add('thumbnail', null, ['label' => 'blog_bundle.property.thumbnail', 'required' => false])
-            ->add('published', null, ['label' => 'blog_bundle.property.published'])
-            ->add('description', 'textarea', ['label' => 'blog_bundle.property.description', 'required' => false]);
+        $formMapper->with('blog_bundle.property.body', ['class' => 'col-md-9'])
+            ->add('title', null, ['label' => 'blog_bundle.property.title'])
+            ->add('slug', null, ['label' => 'blog_bundle.property.slug', 'required' => false])
+            ->add('description', 'textarea', ['label' => 'blog_bundle.property.description', 'required' => false])
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/CategoryAdmin.php]: You have `required=>false` here, and validation constraint `NotBlank`

File: src/Application/BlogBundle/Admin/PostAdmin.php [hunk 2]
```diff
@@ -18,7 +18,7 @@ protected function configureFormFields(FormMapper $formMapper)
     {
         $formMapper->with('blog_bundle.property.body', ['class' => 'col-md-9'])
             ->add('title', null, ['label' => 'blog_bundle.property.title'])
-            ->add('slug', null, ['label' => 'blog_bundle.property.slug'])
+            ->add('slug', null, ['label' => 'blog_bundle.property.slug', 'required' => false])
             ->add('description', 'textarea', ['label' => 'blog_bundle.property.description', 'required' => false])
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/PostAdmin.php]: You have `required=>false` here, and validation constraint `NotBlank`

**Comment by aivus** [comment]
I still getting truncate for `description` field in category

**Comment by aivus** [comment]
What happens if I load file with very long name?

File: src/Application/BlogBundle/Admin/CategoryAdmin.php [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 namespace Application\BlogBundle\Admin;
 
+use Application\BlogBundle\Form\Type\ThumbnailType;
```
Comment (aivus) [inline src/Application/BlogBundle/Admin/CategoryAdmin.php]: Group uses

Comment (Madlexx) [inline src/Application/BlogBundle/Admin/CategoryAdmin.php]: Grouped.

**Comment by Madlexx** [comment]
Resolved in #21

**Comment by Madlexx** [comment]
Changed required state for description.

**Comment by aivus** [comment]
> Changed required state for description.
Please, pay attention. I wrote comment to `CategoryAdmin`

**Comment by Madlexx** [comment]
Fixed. Shame(

**Pull request merged**